### PR TITLE
Add actor clock actuator endpoint 

### LINKED ISSUE #5289: Time travel API for Zeebe
**Is your feature request related to a problem? Please describe.**
In a test environment, or when performing an automated test, I need to manipulate the time to test certain aspects of a workflow or Zeebe. For example, to verify that a timer event triggers after the defined duration. 

Currently, it is possible to pass an `ActorClock` in the constructor of the broker. But this is only available when the broker is created directly from code, and not using a Docker image.  

**Describe the solution you'd like**
Zeebe provides an API to manipulate the time of the broker. For example, by exposing an HTTP endpoint.

To prevent misusage, the API could be available only when a specific profile is enabled (e.g. `develop`, `test`, etc.).

**Describe alternatives you've considered**
Expose the API as a gRPC command. But this API is for testing only and should not be used in a production system. So, exposing this to the client may be misleading.

**Additional context**

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

Adds a new actuator `/actuator/clock/` that can control the actor clock if the config `zeebe.clock.controlled` is set to true.

## Related issues

closes #5289

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java
Comment (npepinpe) [inline dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java:24]: :+1: I like the use of the factory instead of casting the body, the error message is much nicer. It shows good forethought on your part. We should do this more often where applicable.

File: dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java
Comment (npepinpe) [inline dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java:73]: :upside_down_face: Using the `Instant` variant (e.g. `assertThat(body.instant).isCloseTo(Instant.now(), byLessThan(offset + 1, ChronoUnit.MILLIS))`) gives a nicer error message in my opinion (e.g. expected TIME A to be within 10 millis of TIME B), but both are equally readable, so I'll leave it up to you.

File: dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java
Comment (npepinpe) [inline dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java:74]: :question: Could this risk being flaky? Also wondering if maybe it makes sense to check a lower bound?

File: dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java
Comment (npepinpe) [inline dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java:86]: :thought_balloon: I wonder if even without the `sleep` we could guarantee at least one milli expires? I guess this relies on the clock being very accurate, which who knows.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:44]: :x: I don't think this is necessary if you don't use the `@Container` annotation after.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:76]: :wrench: Wouldn't just deploying be enough? What is creating a process instance testing additionally? Same for the test below.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:94]: :wrench: This is coupling the test to the output of the engine. A think slightly more stable version would be to just deploy a process, then await until the number of records doesn't change (i.e. the engine is idle), and compare them. Same for the test below.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:147]: :upside_down_face:  Out of habit I would try to avoid return null unless you explicitly want to deal with null, and instead return `Collections.emptyList()` just in case. I know here it doesn't make much difference, but returning null has different semantics.

File: test-util/src/main/java/io/camunda/zeebe/test/util/actuator/ClockActuatorClient.java
Comment (npepinpe) [inline test-util/src/main/java/io/camunda/zeebe/test/util/actuator/ClockActuatorClient.java:47]:
```suggestion
  private void sendRequest(final String method, final String endpoint, final RequestDto requestDto)
```

File: test-util/src/main/java/io/camunda/zeebe/test/util/actuator/ClockActuatorClient.java
Comment (npepinpe) [inline test-util/src/main/java/io/camunda/zeebe/test/util/actuator/ClockActuatorClient.java:26]: :upside_down_face:  The zeebe prefix seems redundant

File: test-util/src/main/java/io/camunda/zeebe/test/util/actuator/ClockActuatorClient.java
Comment (npepinpe) [inline test-util/src/main/java/io/camunda/zeebe/test/util/actuator/ClockActuatorClient.java:23]: :wrench: Can be final

Comment (npepinpe) [inline test-util/src/main/java/io/camunda/zeebe/test/util/actuator/ClockActuatorClient.java:23]: :x: Event if it's not a public API,  this is meant to be used as an API by other developers, so please add some Javadoc to the class and the public methods. This will be helpful to the next person who needs to use this as well as possibly maintain it.

**Review by npepinpe — CHANGES_REQUESTED** [review]
:rocket: 

Looks really good!

**Comment by oleschoenburg** [comment]
@npepinpe I think there is a bug in `ControlledActorClock` where it doesn't allow negative offsets. `ControlledActorClock.usesOffset` returns false for negative offsets. I only found that now because I was rewriting one of the tests.

File: dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java
Comment (oleschoenburg) [reply dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java:92]: At least locally I can't always produce 1ms difference so sleeping a little bit is safer.

**Comment by npepinpe** [comment]
Looking at `ControlledActorClock`, it looks like it has a few issues, we can do a follow PR to touch on that.

File: dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java
Comment (oleschoenburg) [reply dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java]: I've rewritten it a bit and added a comment clarifying under what conditions this is flaky. tl;dr: it's very unlikely to be flaky.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:44]: :upside_down_face: Can be package private and final (in this case final isn't that important but it's a good habit to indicate intent)

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:59]: :upside_down_face: Maybe we can even start the containers in parallel? :thinking: e.g. `Stream.of(elasticsearchContainer, zeebeContainer).parallel().forEach(Startable::start);` or some such. Zeebe should be able to cope with Elastic not being there.

Comment (npepinpe) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:59]: Similarly I guess we can close them in parallel, but in this case we probably don't gain that much time anyway.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:176]: :upside_down_face: I think there's a way to combine the `waitForExportedRecords` and the extra `searchExportedRecords` in the tests, since Awaitility offers a `<T> T Awaitility.await().until(Callable<T>, Predicate<T>)` variant which would return the latest results of the callable which you could return here. But maybe it's more complicated to use that way, and this is fine.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:41]: :x: I also make this mistake all the time, we should import the real Awaitility and not the shaded one from Testcontainers :smile:

**Review by npepinpe — APPROVED** [review]
:+1: :tada: 

Only one change (the Awaitility import), but nothing that needs another review. 

:thought_balloon: Looking at the actuator clients, I'm thinking we might want to introduce OpenFeign to our code base. It makes these clients much nicer to write, and the error handling is :ok_hand:  (chef's kiss? there's no emoji for it :disappointed:)

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (oleschoenburg) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:176]: Yeah I've tried that first but it got a bit messy

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java
Comment (oleschoenburg) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java:41]: Ack, damn auto-import

**Comment by oleschoenburg** [comment]
I'll rebase this now and then merge. Thanks for the thorough review!

**Comment by oleschoenburg** [comment]
bors merge

**Pull request merged**
